src/SourceLocator/Type/AutoloadSourceLocator.php

```diff
@@ -183,31 +183,30 @@ public function stream_open($path, $mode, $options, &$opened_path) : bool
      * Must be implemented to return some data so that calls like is_file will work.
      *
      * @param string $path
-     * @param int    $flags
+     * @param int    $flags|false
      * @return mixed[]
      * @see https://php.net/manual/en/class.streamwrapper.php
      * @see https://php.net/manual/en/streamwrapper.url-stat.php
      * @phpcsSuppress SlevomatCodingStandard.TypeHints.TypeHintDeclaration.MissingParameterTypeHint
      */
-    public function url_stat($path, $flags) : array
+    public function url_stat($path, $flags)
     {
-        // This is just dummy file stat data to fool stat calls
-        $assoc = [
-            'dev' => 2056,
-            'ino' => 19679399,
-            'mode' => 33204,
-            'nlink' => 1,
-            'uid' => 1000,
-            'gid' => 1000,
-            'rdev' => 0,
-            'size' => 1,
-            'atime' => time(),
-            'mtime' => time(),
-            'ctime' => time(),
-            'blksize' => 4096,
-            'blocks' => 8,
-        ];
-
-        return array_merge(array_values($assoc), $assoc);
+        stream_wrapper_restore('file');
+
+        if ($flags & STREAM_URL_STAT_QUIET) {
+            set_error_handler(function () {
+                // Use native error handler
+                return false;
+            });
+            $result = @stat($path);
+            restore_error_handler();
+        } else {
+            $result = stat($path);
+        }
+
+        stream_wrapper_unregister('file');
+        stream_wrapper_register('file', self::class);
```

Comment:
The code seems to restore the original wrapper, and then set this one again: perhaps the `stat()` approach allows for getting rid of the override in other code locations as well?

Reply:
I'm not sure.

The only other function overriden is:

```php
    public function stream_open($path, $mode, $options, &$opened_path) : bool
    {
        self::$autoloadLocatedFile = $path;
        return false;
    }
```

If makes sense to return false, because the call to "stream_open" is triggered by `include "the_file.php"`. If we used the same approach for "stream_open", the file would be really included. If there is a syntax error in the PHP file, PHP will crash. But with the current approach, the file is not included and therefore, syntax errors are analyzed by PHP-parser and throws a nice exception.

Reply:
Done. Tried to explain the process in the docblock: https://github.com/moufmouf/BetterReflection/blob/2d0ea4440eb82611f26cf972bfbfe71c03dd4306/src/SourceLocator/Type/AutoloadSourceLocator.php#L183-L185
Is that good enough?

Reply:
:100: